### PR TITLE
Update storybook with adhoc calling page

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What
Add Adhoc calling conceptual docs page to storybook. 
This PR also includes: 
- a code snippet to show how to customize display name and avatars in the context of adhoc calling.
- code snippet fixes

# Why
To provide Contoso with knowledge about Adhoc calling.

# How Tested
Tested in local storybook

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by dmceachernmsft — APPROVED** [review]

File: change/@azure-communication-react-7d52f99e-a511-4b68-9e24-d63abb254863.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Added adhoc calling conceptual docs page in storybook",
```
Comment (edwardlee-msft) [inline change/@azure-communication-react-7d52f99e-a511-4b68-9e24-d63abb254863.json:3]:
```suggestion
  "comment": "Add adhoc calling conceptual docs page to storybook",
```

File: packages/storybook/stories/AdhocCalling.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+added to the `CallParticipantsLocator` which is used as the locator in the
+[AzureCommunicationCallAdapterArgs](./?path=/docs/composite-adapters--page#azurecommunicationcalladapterargs)
+when creating your [CallAdapter](path=/docs/composite-adapters--page#call-adapter).
+In the current beta version, only one Teams user can be called. A Teams call queue can also be called. For more
+information on call queues, refer to this
+[Learn documentation](https://learn.microsoft.com/en-us/microsoftteams/create-a-phone-system-call-queue)
+
+<MessageBar messageBarType={MessageBarType.warning}>
+  Note: Teams user ids need to be in format '8:orgid:UUID' when used in the CallParticipantsLocator. For example,
+  '8:orgid:87d349ed-44d7-43e1-9a83-5f2406dee5bd'.
```
Comment (edwardlee-msft) [inline packages/storybook/stories/AdhocCalling.stories.mdx:29]: Is this a real mri? Maybe we can switch this out with 80000000-0000000-00000-00 etc.

File: packages/storybook/stories/AdhocCalling.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+The id of a Teams user or a call queue can be obtained by using
+[Microsoft Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer).
+From there, sign in by clicking the circled button as shown below with a Microsoft account in the same tenant as the
+Teams user or call queue you want to call.
+
+<img style={{ width: 'auto', height: 'auto' }} src="images/graph-api-sign-in.png" />
+
+Then create a
+[GET request](https://learn.microsoft.com/en-us/graph/api/user-get?view=graph-rest-1.0&tabs=http#http-request) for
+v1.0 of the users endpoint using the Teams user's userPrincipalName like shown below:
```
Comment (edwardlee-msft) [inline packages/storybook/stories/AdhocCalling.stories.mdx:57]:
```suggestion
v1.0 of the users endpoint using the Teams user's `userPrincipalName` like shown below:
```

File: packages/storybook/stories/AdhocCalling.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+<img style={{ width: 'auto', height: 'auto' }} src="images/graph-api-get-user-id-api-request.png" />
+
+If you want to obtain the id for a call queue, use the userPrincipalName of Teams resource account linked to the call
```
Comment (edwardlee-msft) [inline packages/storybook/stories/AdhocCalling.stories.mdx:61]:
```suggestion
If you want to obtain the id for a call queue, use the `userPrincipalName` of Teams resource account linked to the call
```

**Review by edwardlee-msft — APPROVED** [review]

File: packages/storybook/stories/AdhocCalling.stories.mdx (hunk 2 above)
Comment (mgamis-msft) [reply packages/storybook/stories/AdhocCalling.stories.mdx:29]: It is an mri from the Sample tenant.
image

**Pull request merged**